src/robot/running/keywordrunner.py

```diff
@@ -206,16 +256,99 @@ def _to_number_with_arithmetics(self, item):
             raise TypeError("Expected number, got %s." % type_name(item))
         return number
 
-    def _run_one_round(self, data, values):
-        name = ', '.join(format_assign_message(var, item)
-                         for var, item in zip(data.variables, values))
-        result = KeywordResult(kwname=name,
-                               type=data.FOR_ITEM_TYPE)
-        for var, value in zip(data.variables, values):
-            self._context.variables[var] = value
-        runner = KeywordRunner(self._context, self._templated)
-        with StatusReporter(self._context, result):
-            runner.run_keywords(data.keywords)
+
+class ForInZipLoopRunner(BasicForLoopRunner):
+    def __init__(self, context, templated=False):
+        super(ForInZipLoopRunner, self).__init__(context, templated)
+
+    def _flavor_name(self):
+        return 'IN ZIP'
+
+    def _replace_variables(self, data):
+        """
+        Make sure we have as many loop-variables as zipped-iterables.
+
+        (Other than that, delegate to the superclass.)
+        """
+        values = super(ForInZipLoopRunner, self)._replace_variables(data)
+        if len(data.variables) == len(data.values):
+            return values
+        raise DataError("Expected %d Loop variables, but found %d"
```

Comment:
- `loop` instead of `Loop`
- `got` is used instead of `found` with other errors. I'd prefer that also here for consistency reasons.
- We generally have `.` at the end of the error when it is a full sentence.
- The error could be more explicit. Perhaps something like `'FOR IN ZIP expects equal number of variables and iterables. Got %d variable%s and %d iterable%s.' % (len(data.variables), s(data.variables), len(data.values), s(data.values)`.